development/extensions/tutorial.rst

```diff
@@ -0,0 +1,297 @@
+   time is optional)"
+   ``license`` | The license must be ``GPL-2.0`` for now
+   ``authors`` | "An array with the authors of the extension.
+   See `authors`_ for more details."
+   ``require`` | "An array with requirements of the extension.
+   See `require`_ for more details."
+   ``require-dev`` | "An array with development requirements of the extension.
+   See `require-dev`_ for more details."
+   ``extra`` | "An array with additional values of the extension.
+   See `extra`_ for more details."
+
+authors
+-------
+
+You may have unlimited authors. But you should at least have one.
+
+.. csv-table::
+   :header: "Field", "Content"
+   :delim: |
+
+   ``name`` | Name of the author
+   ``email`` | Email address of the author
+   ``homepage`` | Must contain one valid URL or be empty
+   ``role`` | "Role can be used to specify what the authors did for the
+   extension (e.g. Developer, Translator, Supporter, ...)"
+
+require
+-------
+
+In the ``require`` section you can specify technical requirements of your
+extension. Examples are the ``php`` version, or
+`third party libraries <https://packagist.org/>`_. Since our demo extension does
+not require any additional library, we only use the php version requirement, to
+make sure people have the right php version on their server. phpBB 3.1 requires
+php 5.3.3 or higher, so the version comparison is ``>= 5.3.3``.
+
+require-dev
+-----------
+
+In the ``require-dev`` section you can specify technical requirements of your
+extension, which are only required for development. We use the
+`Extension PreValidation <https://packagist.org/packages/phpbb/epv>`_ Tool from
+the phpBB Extension team here. To perform some basic validation when running our
+tests on Travis CI later. Since we always want to have the newest version, we
+require ``dev-master``.
+
+.. todo:: add link to testing/travis section.
+
+extra
+-----
+
+This section contains only additional information and is up to free usage in
+terms of the composer-specification. However, phpBB is using 2 special entries
+in this array for extensions:
+
+.. csv-table::
+   :header: "Field", "Content"
+   :delim: |
+
+   ``display-name`` | "Display name of the extension (can be different then the
+   folder name)"
+   ``soft-require`` | "Soft requirements are basically like `require`_. The only
+   difference is, that composer does not know that these requirements exist.
+   This allows us e.g. to compare the phpBB version, although there might not be
+   a phpBB package with the specified version. In this case we require any 3.1
+   version. This can be done, by prefixing it with a ``~``:
+   ``""phpbb/phpbb"": ""~3.1""``"
+
+Enable extension
+================
+
+After you filled the ``composer.json`` with the content from above, you can go
+to the "Administration Control Panel" (ACP) > "Customise" > "Extensions" and
+enable the extension.
+
+HTML Events
+===========
+
+Up to now, we have an extension that does not. So let's start with adding some
```

Comment:
"So far, all we have done is create an extension that does nothing. So let's create some code to generate output and verify that the extension is working."